AWSCore/Networking/AWSURLSessionManager.m

```diff
@@ -323,7 +323,12 @@ - (void)URLSession:(NSURLSession *)session task:(NSURLSessionTask *)sessionTask
                                                                                                         data:delegate.responseData
                                                                                                        error:&error];
                     if (error) {
-                        delegate.error = error;
+                        if([delegate.responseObject isKindOfClass:[NSDictionary class]]) {
+                            delegate.error = [NSError errorWithDomain:AWSNetworkingErrorDomain code:AWSNetworkingErrorUnknown userInfo: ((NSDictionary *)delegate.responseObject)[@"Error"]];
```

Comment:
We should persist the error domain & code from the original error, to make this a purely additive change

```objc
if ([delegate.responseObject isKindOfClass:[NSDictionary class]]) {
    delegate.error = [NSError errorWithDomain:error.domain
                                         code:error.code
                                     userInfo:((NSDictionary *)delegate.responseObject)[@"Error"]];
}
```